src/lib.rs

```diff
@@ -14,19 +14,193 @@ use rustty::CellAccessor;
+
+    #[test]
+    fn test_bitvec_bit_order() {
+        // Consider a two-byte number where the firt byte's value is 10 and
+        // the second byte's value is 7. Converting to a little endian byte
+        // array should make the 0th byte 10 and the 1th byte 7.
+        let ten_seven = BigInt::from(10 + 7*(2.pow(8)));
+        let bytes = ten_seven.to_bytes_le().1;
+        assert_eq!(bytes[0], 10);
+        assert_eq!(bytes[1], 7);
+        // Now check what happens when we convert this to a BitVec. The bytes
+        // are in little endian order, but the bits within each byte are big
+        // endian:
+        // [00001010][00000111]
+        let bits = BitVec::from_bytes(&bytes);
+        let expected = vec![
+            // 0th byte representing 10
+            false,
+            false,
+            false,
+            false,
+            true,
+            false,
+            true,
+            false,
+            // 1st byte representing 7
+            false,
+            false,
+            false,
+            false,
+            false,
+            true,
+            true,
+            true];
+        for i in 0..16 {
+            assert_eq!(bits[i], expected[i]);
+        }
+    }
+
+    #[test]
+    fn test_conway_kode() {
+        let expected = "476348294852520375132009738840824718882889556\
+                        423255282629108876378472743729817205343700177\
+                        683429960362194923168607044012736510546282236\
+                        08960".parse::<BigInt>().unwrap();
+        assert_eq!(expected, conway_kode());
+    }
+
+    #[test]
+    fn test_grid_from_gidney_string() {
+
+        let grid = grid_from_gidney_string(vec!["   ", "   "]);
+        let mut expected = CellSet::new();
+        assert_eq!(grid, expected);
+
+        let grid = grid_from_gidney_string(vec!["#  ", "   "]);
+        expected.insert((0, 0));
+        assert_eq!(grid, expected);
+
+        let grid = grid_from_gidney_string(vec!["#  ", " # "]);
+        expected.insert((1, 1));
+        assert_eq!(grid, expected);
+    }
+
+    #[test]
+    fn test_decimal_encoded_string_to_bitvec() {
+        let s = "1802";
+        let bitvec = decimal_encoded_string_to_bitvec(s);
+        for i in 0..16 {
+            assert_eq!(bitvec[i], EXPECTED_1082_BITS[i]);
+        }
+    }
+
+    #[test]
+    fn test_bigint_to_bitvec(){
+        let v = bigint_to_bitvec(BigInt::from(1802));
+        for i in 0..16 {
+            assert_eq!(v[i], EXPECTED_1082_BITS[i]);
+        }
+    }
+
+    #[test]
+    fn test_get_state() {
+        let mut grid = CellSet::new();
+        //  0
+        // 0#< look here
+        //  ^
+        grid.insert((0, 0));
+        assert_eq!(get_state(&grid, &(0, 0)), 16); // 2^4
+        //  01
+        // 0#-< look here
+        // 1-#
+        //  ^
+        grid.insert((1, 1));
+        assert_eq!(get_state(&grid, &(0, 0)), 272); // 2^4 + 2^8
+    }
+}
+
+pub fn conway_kode() -> BigInt {
```

Comment:
This should be moved into test and made private. If we need it later we can move it back into the main library.

Reply:
I think it's useful to have the rule for Conway stored somewhere accessible to the outside. I propose we remove this function entirely and instead provide a public decimal string representation constant of the Conway Kode.